tests/unittests/unittests.cpp

```diff
@@ -4264,6 +4269,137 @@ class ConcurrentQueueTests : public TestClass<ConcurrentQueueTests>
 		
 		return true;
 	}
+
+	bool acquire_and_signal()
+	{
+		typedef ::moodycamel::LightweightSemaphore semaphore;
+
+		// Test resource acquisition from one other thread.
+		{
+			semaphore s;
+			s.signal(); // Single resource available
+						
+			const auto fnTestSingleAcquire = 
+				[](::moodycamel::LightweightSemaphore &s) -> bool
+				{
+					for (std::size_t k = 0; k < 100; ++k)
+					{
+						ASSERT_OR_FAIL(s.wait(100 * 1000)); // Wait 100 ms
```

Comment:
Hmm, not sure I like this. It could technically fail if the thread is starved for more than 100ms. Can we just wait with an infinite timeout?